server/sonar-server/src/main/java/org/sonar/server/platform/Platform.java

```diff
@@ -78,14 +82,29 @@ public void init(Properties properties, ServletContext servletContext) {
 
   // Platform is injected in Pico, so do not rename this method "start"
   public void doStart() {
-    if (!started && getDatabaseStatus() == DatabaseVersion.Status.UP_TO_DATE) {
+    if (started && !isInSafeMode()) {
+      return;
+    }
+
+    if (getDatabaseStatus() == DatabaseVersion.Status.UP_TO_DATE) {
       startLevel34Containers();
-      started = true;
+      if (isInSafeMode()) {
```

Comment:
I find safemode appropriate because here, it's more a matter of describing the state in which the SQ instance is than describing the reason it is in this state.

The reason it is in this state might evolve over time.
eg. SQ instance could start even if when a plugin problem occurs at startup, this way avoiding having an instance down. This is especially something we may want if we implement the plugin update evolution we talked about.